bal/bal.c

```diff
@@ -152,18 +234,47 @@ static int baldev_release(struct inode *inode, struct file *filp)
 static long
 baldev_ioctl(struct file *filp, unsigned int cmd, unsigned long arg)
 {
-	int status = -EINVAL;
-	if (cmd == BAL_IOC_BUSY_PIN) {
-		if (gpio_is_valid(arg)) {
-			status = gpio_request(arg, "BUSY pin");
-			if (!status) {
-				dev_info(&bal.spi->dev, "Setting BUSY pin to %d\n", (unsigned int)arg);
-				gpio_free(bal.busy_pin);
-				bal.busy_pin = (unsigned int)arg;
-				gpio_direction_input(bal.busy_pin);
+	//printk(KERN_ERR "%s : %d fp=%ld cmd=%ld arg=%lx\n", __FUNCTION__, __LINE__, (unsigned long int)filp, cmd, arg);
+	printk(KERN_ERR "my ioctl - cmd: %d - arg: %lu\n", cmd, arg);
```

Comment:
This is for debugging purpose, so KERN_ERR shouldn't be used.
Moreover, one the device is successfully probed and driver is loaded, printk should also not be used.
Please replace them with
dev_err or dev_info or dev_dbg

Example: dev_dbg(&(bal.spi->dev), "my ioctl - cmd: %d - arg: %lu\n", cmd, arg);